core/src/main/java/org/elasticsearch/bootstrap/ElasticsearchUncaughtExceptionHandler.java

```diff
@@ -71,7 +72,17 @@ static boolean isFatalUncaught(Throwable e) {
     }
 
     private static boolean isFatalCause(Throwable cause) {
-        return cause instanceof Error;
+        // We wrap AlreadyClosedException in AssertionError when we re-throw to ensure catch clauses don't suppress it, but we should not
+        // halt the JVM for that case:
+        return cause instanceof Error && isWrappedAlreadyClosedException(cause) == false;
```

Comment:
why?

Please, let it halt the jvm. the jvm will halt in a much nastier way if we do not resolve the bugs around this :)